src/main/java/ch/uzh/ifi/hase/soprafs23/service/UserService.java

```diff
@@ -208,4 +211,20 @@ public void addUserIngredients(Long userId, List<IngredientPutDTO> ingredientsPu
         user.addIngredient(newIngredients);
         userRepository.save(user);
     }
+
+    public void leaveGroup(Long userId) {
+        User user = getUserById(userId);
+        Long groupId = user.getGroupId();
+        if (groupId != null) {
+            Group group = groupRepository.findById(groupId).orElseThrow(() ->
+                    new ResponseStatusException(HttpStatus.NOT_FOUND, String.format("Group with id %d does not exist", groupId)));
```

Comment:
I think it'd be better if you let UserService have access to GroupService instead of GroupRepository.
Then Group group = getGroupById(groupId); would do this 404 for you (no need for the orElseThrow()), and also we would have the Repositories only being user by their corresponding Services -> a bit less coupling, and the Services have all the functionality needed